include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
True. And `sizeof(mode_enums) / sizeof(mode_enums[0])` is even more correct.

And `sizeof(mode_enums) / sizeof(char *)` still more correct because C allows empty arrays to be declared.

Also, there's no need for specialized functions at all. You can use a macro that takes
a name and picks out the right string array:

```
static const char *ss_enumA[];

#define ENUM_STR(NAME, S) parse (ss_enum ## NAME, sizeof(ss_enum ## NAME) / sizeof(char *), S)

enum A val = ENUM_STR (A, "FOOBAR");
```